syntax = "proto3";

package systeminsight.proto;

option cc_enable_arenas = true;

message MetricLabel {
  string key = 1;
  string value = 2;
}

message MetricSample {
  string name = 1;
  double value = 2;
  int64 timestamp_ms = 3;
  repeated MetricLabel labels = 4;
}

message MetricsReport {
  repeated MetricSample samples = 1;
  string host_id = 2;
  string collector_version = 3;
}

message ReportAck {
  bool ok = 1;
  string message = 2;
}

service SystemInsightService {
  rpc SendMetrics(MetricsReport) returns (ReportAck);
}

